src/gsy_e/models/state.py

```diff
@@ -673,7 +700,10 @@ def market_cycle(self, past_time_slot, current_time_slot: DateTime, all_future_t
         Simulate actual Energy flow by removing pledged storage and adding bought energy to the
         used_storage
         """
-        self.add_default_values_to_state_profiles(all_future_time_slots)
+        # In case the future market is enabled, the current time_slot will already be populated
+        # So we need to reset this time_slot values to defaults
+        self._delete_time_slot(current_time_slot)
+        self.add_default_values_to_state_profiles([current_time_slot, *all_future_time_slots])
```

Comment:
Maybe we can save the effort of both steps here with for non-enabled future markets. What do you think?

Reply:
Makes sense, I disabled this process for simulations that don't enable future markets